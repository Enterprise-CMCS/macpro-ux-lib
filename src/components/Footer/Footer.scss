a > svg {
  padding-top: 10px;
}

@media only screen and (max-width: 640px) {
  .mobile-text-center {
    text-align: center;
  }
}
