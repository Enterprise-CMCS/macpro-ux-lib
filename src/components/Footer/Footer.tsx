import React from "react";
import { Logo } from "../Logo/Logo";
import depHealthHumanServicesWhite from "../../assets/img/logos/dept_health_human_services_usa_white.svg";
import depHealthHumanServicesBlack from "../../assets/img/logos/dept_health_human_services_usa_black.svg";
import medicaidLogoWhite from "../../assets/img/logos/medicaid_gov_white.svg";
import medicaidLogoBlack from "../../assets/img/logos/medicaid_gov_black.svg";
import { Icon, IconChoice } from "../Icon/Icon";
type IntrinsicElements = JSX.IntrinsicElements["footer"];

interface Link {
  href?: string;
  iconName?: IconChoice;
  onClick?: () => any;
  text: string;
  target?: string;
}

interface Props extends IntrinsicElements {
  emailAddress: string;
  address?: string;
  altFooter?: boolean;
  navigationLinks?: Link[];
}

/**
 * Footer Component
 * @param {string}                        emailAddress       Email Address used at the bottom left for contact help.
 * @param {string}                        [address]          Address used at the bottom right of footer.
 * @param {boolean}                       [altFooter]        Determines whether to use the alternative footer.
 * @param {Link[]}                        [navigationLinks]  Array of navigation links to render in the alternative footer.
 */

export const Footer: React.FC<Props> = ({
  altFooter = false,
  emailAddress,
  address = "7500 Security Boulevard Baltimore, MD 21244",
  navigationLinks,
  ...rest
}) => {
  return altFooter ? (
    <footer className="usa-footer text-white" {...rest}>
      <div className="usa-footer__primary-section bg-primary">
        <div className="grid-row padding-x-5 padding-y-2 flex-align-center">
          <div className="grid-col text-white">
            {navigationLinks && (
              <div className="grid-col-4">
                <h4 className="margin-0">Sitemap</h4>
                <hr />
                <div className="grid-row">
                  {navigationLinks.map((link, idx) => {
                    return (
                      <div key={idx} className="grid-col-6">
                        <a
                          className="text-white font-sans-md text-no-underline padding-x-1 grid-col"
                          href={link.href}
                          onClick={link.onClick}
                          target={link.target}
                        >
                          {link.text}
                          {link.iconName && (
                            <Icon
                              name={link.iconName}
                              color="#fff"
                              className="margin-top-3"
                            />
                          )}
                        </a>
                      </div>
                    );
                  })}
                </div>
              </div>
            )}
          </div>

          <div className="grid-col-4">
            <div className="grid-row flex-align-center">
              <Logo
                ariaLabel="Medicaid Logo"
                altText="Medicaid Logo"
                source={medicaidLogoWhite}
              />
              <Logo
                ariaLabel="Department of Health and Human Services"
                altText="Department of Health and Human Services Logo"
                source={depHealthHumanServicesWhite}
                className="grid-col-3 margin-left-1"
                height={80}
              />
            </div>
            <div className="font-sans-xs">
              A federal government website managed and paid for by the U.S.
              Centers for Medicare and Medicaid Services and part of the MACPro
              suite.
            </div>
          </div>
        </div>
      </div>
      <div className="usa-footer__secondary-section bg-primary-darker text-white font-sans-2xs padding-x-4">
        <div className="grid-row">
          <p className="tablet:grid-col-9 mobile-text-center">
            Email {<b className="text-underline">{emailAddress}</b>} for help or
            feedback.
          </p>

          <p className="tablet:grid-col-3 mobile-text-center">{address}</p>
        </div>
      </div>
    </footer>
  ) : (
    <footer className="usa-footer usa-footer--slim" {...rest}>
      <div className="usa-footer__primary-section bg-accent-cool-lighter flex-justify-end">
        <div className="grid-row padding-x-5 padding-y-2 flex-align-center">
          <div className="tablet:grid-col mobile-text-center">
            <Logo
              ariaLabel="Medicaid Logo"
              altText="Medicaid Logo"
              source={medicaidLogoBlack}
            />
          </div>

          <div className="grid-row tablet:grid-col-5 flex-align-center">
            <Logo
              ariaLabel="Department of Health and Human Services"
              altText="Department of Health and Human Services Logo"
              source={depHealthHumanServicesBlack}
              className="grid-col tablet:grid-col-3 text-center"
              height={80}
            />
            <p className="grid-col-12 tablet:grid-col-9 font-sans-xs">
              A federal government website managed and paid for by the U.S.
              Centers for Medicare and Medicaid Services and part of the MACPro
              suite.
            </p>
          </div>
        </div>
      </div>
      <div className="usa-footer__secondary-section bg-primary text-white font-sans-2xs padding-x-4">
        <div className="grid-row">
          <p className="tablet:grid-col-9 mobile-text-center">
            Email {<b className="text-underline">{emailAddress}</b>} for help or
            feedback.
          </p>
          <p className="tablet:grid-col-3 mobile-text-center">{address}</p>
        </div>
      </div>
    </footer>
  );
};
