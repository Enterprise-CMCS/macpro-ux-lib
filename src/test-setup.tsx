import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import "@testing-library/jest-dom";

// Here we can set all providers for themes, stores and other stuff
const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  return <>{children}</>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
