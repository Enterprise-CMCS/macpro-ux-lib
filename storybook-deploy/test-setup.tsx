import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import "@testing-library/jest-dom";

// Here we can set all providers for themes, stores and other stuff
const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  return <>{children}</>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };

/**
 * Clean up attributes from the rendered component by removing the generated ids.
 *
 * Before:
 *   for="input-type-text-123456"
 *   id="input-type-text-123456"
 *
 * After:
 *   for="input-type-text"
 *   id="input-type-text"
 *
 * This is useful during snapshot testing. Snapshot comparisons will fail without addressing these items.
 *
 * @param {HTMLElement} container The component to clean up.
 * @param {prefixes}      string[]  The reference substrings to leave alone.
 */
