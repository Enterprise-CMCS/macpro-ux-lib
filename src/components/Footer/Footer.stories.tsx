import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { Footer } from "./Footer";

export default {
  title: "USWDS/Base/Footer",
  component: Footer,
  argTypes: {
    emailAddress: {
      description: "Email Address used at the bottom left for contact help",
    },
    address: {
      description: "Address used at the bottom right of footer",
    },
    altFooter: {
      description: "Determines whether to use the alternative footer",
    },
    navigationLinks: {
      description: "Array of navigation elements to render in footer",
    },
  },
  args: {
    emailAddress: "sample@cms.hhs.gov",
    address: "7500 Security Boulevard Baltimore, MD 21244",
    altFooter: false,
  },
} as ComponentMeta<typeof Footer>;

const Template: ComponentStory<typeof Footer> = ({ ...rest }) => (
  <Footer {...rest} />
);

export const Default = Template.bind({});
export const altFooter = Template.bind({});
altFooter.args = {
  altFooter: true,
  navigationLinks: [
    <a href="https://www.google.com/">test</a>,
    <a href="https://www.google.com/">test</a>,
    <a href="https://www.google.com/">test</a>,
    <a href="https://www.google.com/">test</a>,
    <a href="https://www.google.com/">test</a>,
  ],
};
