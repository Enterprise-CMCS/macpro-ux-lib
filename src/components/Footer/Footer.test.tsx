import React from "react";
import { screen, render } from "../../test-setup";
import { Footer } from "./Footer";

describe("Tests for the Footer component.", () => {
  it("Should render", () => {
    render(
      <Footer
        emailAddress="test@gmail.com"
        data-testid="Footer"
        address="test"
      />
    );

    const IconComp = screen.getByTestId("Footer");
    expect(IconComp).toBeInTheDocument();
  });
});

describe("compontent snapshots", () => {
  it("accessibility new icon", () => {
    const { container } = render(
      <Footer emailAddress="test@gmail.com" address="test" />
    );
    expect(container).toMatchSnapshot();
  });
});
