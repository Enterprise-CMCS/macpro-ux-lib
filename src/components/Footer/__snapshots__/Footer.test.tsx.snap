// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compontent snapshots Alternative Footer 1`] = `
<div>
  <footer
    class="usa-footer text-white"
  >
    <div
      class="usa-footer__primary-section bg-primary"
    >
      <div
        class="grid-row padding-x-5 padding-y-2 flex-align-center"
      >
        <div
          class="grid-col text-white"
        >
          <div
            class="grid-col-4"
          >
            <h4
              class="margin-0"
            >
              Sitemap
            </h4>
            <hr />
            <div
              class="grid-row"
            >
              <div
                class="grid-col-6"
              >
                <a
                  class="text-white font-sans-md text-no-underline padding-x-1 grid-col"
                  href="https://www.google.com/"
                >
                  FAQ
                  <svg
                    aria-hidden="true"
                    aria-label="launch icon"
                    class="usa-icon--size-3"
                    color="#fff"
                    fill="currentColor"
                    role="img"
                  >
                    <use
                      href="sprite.svg#launch"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="grid-col-4"
        >
          <div
            class="grid-row flex-align-center"
          >
            <div
              aria-label="Medicaid Logo"
              class="application-logo-container "
              role="img"
            >
              <img
                alt="Medicaid Logo"
                aria-hidden="false"
                src="[object Object]"
                width="200"
              />
            </div>
            <div
              aria-label="Department of Health and Human Services"
              class="application-logo-container grid-col-3 margin-left-1"
              role="img"
            >
              <img
                alt="Department of Health and Human Services Logo"
                aria-hidden="false"
                height="80"
                src="[object Object]"
                width="200"
              />
            </div>
          </div>
          <div
            class="font-sans-xs"
          >
            A federal government website managed and paid for by the U.S. Centers for Medicare and Medicaid Services and part of the MACPro suite.
          </div>
        </div>
      </div>
    </div>
    <div
      class="usa-footer__secondary-section bg-primary-darker text-white font-sans-2xs padding-x-4"
    >
      <div
        class="grid-row"
      >
        <p
          class="tablet:grid-col-9 mobile-text-center"
        >
          Email
           
          <a
            href="mailto:test@gmail.com"
          >
            <b>
              test@gmail.com
            </b>
          </a>
           
          for help or feedback.
        </p>
        <p
          class="tablet:grid-col-3 mobile-text-center"
        >
          111 test drive
        </p>
      </div>
    </div>
  </footer>
</div>
`;

exports[`compontent snapshots Basic Footer 1`] = `
<div>
  <footer
    class="usa-footer usa-footer--slim"
  >
    <div
      class="usa-footer__primary-section bg-accent-cool-lighter flex-justify-end"
    >
      <div
        class="grid-row padding-x-5 padding-y-2 flex-align-center"
      >
        <div
          class="tablet:grid-col mobile-text-center"
        >
          <div
            aria-label="Medicaid Logo"
            class="application-logo-container "
            role="img"
          >
            <img
              alt="Medicaid Logo"
              aria-hidden="false"
              src="[object Object]"
              width="200"
            />
          </div>
        </div>
        <div
          class="grid-row tablet:grid-col-5 flex-align-center"
        >
          <div
            aria-label="Department of Health and Human Services"
            class="application-logo-container grid-col tablet:grid-col-3 text-center"
            role="img"
          >
            <img
              alt="Department of Health and Human Services Logo"
              aria-hidden="false"
              height="80"
              src="[object Object]"
              width="200"
            />
          </div>
          <p
            class="grid-col-12 tablet:grid-col-9 font-sans-xs"
          >
            A federal government website managed and paid for by the U.S. Centers for Medicare and Medicaid Services and part of the MACPro suite.
          </p>
        </div>
      </div>
    </div>
    <div
      class="usa-footer__secondary-section bg-primary text-white font-sans-2xs padding-x-4"
    >
      <div
        class="grid-row"
      >
        <p
          class="tablet:grid-col-9 mobile-text-center"
        >
          Email
           
          <a
            href="mailto:test@gmail.com"
          >
            <b>
              test@gmail.com
            </b>
          </a>
           
          for help or feedback.
        </p>
        <p
          class="tablet:grid-col-3 mobile-text-center"
        >
          7500 Security Boulevard Baltimore, MD 21244
        </p>
      </div>
    </div>
  </footer>
</div>
`;
